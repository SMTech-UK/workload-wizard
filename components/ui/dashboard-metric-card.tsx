import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-500">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-muted-foreground mt-1">{subtitle}</div>}
    </div>
  );
} 