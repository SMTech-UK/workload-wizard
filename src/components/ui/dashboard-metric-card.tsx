import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white dark:bg-zinc-900 shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-900 dark:text-white">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-gray-600 dark:text-gray-300 mt-1">{subtitle}</div>}
    </div>
  );
} 