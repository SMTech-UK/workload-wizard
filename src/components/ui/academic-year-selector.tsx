"use client"

import React from "react";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { Badge } from "@/components/ui/badge";
import { useAcademicYear } from "@/hooks/useAcademicYear";
import { Calendar } from "lucide-react";

interface AcademicYearSelectorProps {
  className?: string;
  showActiveBadge?: boolean;
  onAcademicYearChange?: (academicYearId: string) => void;
}

export function AcademicYearSelector({ 
  className, 
  showActiveBadge = true,
  onAcademicYearChange 
}: AcademicYearSelectorProps) {
  const { 
    academicYears, 
    currentAcademicYear, 
    currentAcademicYearId, 
    changeAcademicYear 
  } = useAcademicYear();

  const handleAcademicYearChange = async (academicYearId: string) => {
    try {
      await changeAcademicYear(academicYearId);
      onAcademicYearChange?.(academicYearId);
    } catch (error) {
      console.error('Failed to change academic year:', error);
    }
  };

  // Function to format academic year name (e.g., "2025/26" -> "AY-25/26")
  const formatAcademicYearName = (yearName: string) => {
    // Extract the last 2 digits from each year part
    const parts = yearName.split('/');
    if (parts.length === 2) {
      const startYear = parts[0].slice(-2);
      const endYear = parts[1].slice(-2);
      return `AY-${startYear}/${endYear}`;
    }
    // Fallback to original name if format is unexpected
    return yearName;
  };

  if (!academicYears || academicYears.length === 0) {
    return (
      <div className={`flex items-center gap-2 text-muted-foreground ${className}`}>
        <Calendar className="h-4 w-4" />
        <span className="text-sm">No academic years configured</span>
      </div>
    );
  }

  return (
    <div className={`flex items-center gap-2 ${className}`}>
      <Select
        value={currentAcademicYearId || ""}
        onValueChange={handleAcademicYearChange}
      >
        <SelectTrigger className="w-auto min-w-[120px] flex items-center gap-2">
          <Calendar className="h-4 w-4 text-muted-foreground" />
          <span className="font-medium">
            {currentAcademicYear ? formatAcademicYearName(currentAcademicYear.name) : "Select AY"}
          </span>
        </SelectTrigger>
        <SelectContent>
          {academicYears.map((year: any) => (
            <SelectItem key={year._id} value={year._id}>
              <div className="flex items-center gap-2">
                <Calendar className="h-4 w-4 text-muted-foreground" />
                <span>{formatAcademicYearName(year.name)}</span>
                {year.isActive && showActiveBadge && (
                  <Badge variant="default" className="text-xs">Active</Badge>
                )}
                {year.isStaging && (
                  <Badge variant="secondary" className="text-xs">Staging</Badge>
                )}
              </div>
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
    </div>
  );
} 