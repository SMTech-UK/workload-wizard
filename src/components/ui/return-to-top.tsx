"use client";

import { Button } from "@/components/ui/button";
import { ChevronUp, Home } from "lucide-react";
import { useState, useEffect } from "react";
import Link from "next/link";

export function ReturnToTop() {
  const [isVisible, setIsVisible] = useState(true); // Temporarily always visible for testing

  useEffect(() => {
    const toggleVisibility = () => {
      console.log("Scroll position:", window.pageYOffset);
      if (window.pageYOffset > 300) {
        console.log("Setting visible to true");
        setIsVisible(true);
      } else {
        console.log("Setting visible to false");
        setIsVisible(false);
      }
    };

    // Call once on mount to check initial state
    toggleVisibility();
    
    window.addEventListener("scroll", toggleVisibility);
    return () => window.removeEventListener("scroll", toggleVisibility);
  }, []);

  const scrollToTop = () => {
    console.log("Scrolling to top");
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  console.log("ReturnToTop component render, isVisible:", isVisible);

  return (
    <>
      {/* Return to Home button - top left */}
      <Link href="/">
        <Button
          size="sm"
          variant="outline"
          className="fixed top-4 left-4 z-50 shadow-lg hover:shadow-xl transition-all duration-200"
          aria-label="Return to home"
        >
          <Home className="h-4 w-4 mr-2" />
          Home
        </Button>
      </Link>

      {/* Return to Top button - bottom left */}
      <Button
        onClick={scrollToTop}
        size="sm"
        className="fixed bottom-4 left-4 z-50 rounded-full shadow-lg hover:shadow-xl transition-all duration-200 bg-blue-500 text-white"
        aria-label="Return to top"
      >
        <ChevronUp className="h-4 w-4" />
      </Button>
    </>
  );
} 