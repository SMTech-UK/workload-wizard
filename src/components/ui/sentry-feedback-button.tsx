"use client";

import { useEffect, useState } from "react";
import * as Sentry from "@sentry/nextjs";
import { Button } from "@/components/ui/button";
import { Bug, MessageCircle } from "lucide-react";
import { useModalAwareButton } from "@/hooks/useModalAwareButton";
import { createPortal } from "react-dom";

export function SentryFeedbackButton() {
  const [feedback, setFeedback] = useState<any>(null);
  const [widget, setWidget] = useState<any>(null);
  const [mounted, setMounted] = useState(false);
  const buttonRef = useModalAwareButton();

  useEffect(() => {
    setMounted(true);
    // Get the feedback instance on the client side
    setFeedback(Sentry.getFeedback());
  }, []);

  const handleFeedbackClick = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    
    if (!feedback) return;

    if (widget) {
      // If widget is open, close it
      widget.removeFromDom();
      setWidget(null);
    } else {
      // Create and show the form immediately
      feedback.createForm().then((newWidget: any) => {
        newWidget.appendToDom();
        newWidget.open();
        setWidget(newWidget);
      }).catch((error: any) => {
        console.error('Failed to create Sentry feedback form:', error);
      });
    }
  };

  if (!feedback || !mounted) return null;

  const button = (
    <Button
      ref={buttonRef}
      variant="destructive"
      size="lg"
      onClick={handleFeedbackClick}
      onMouseDown={(e) => {
        e.preventDefault();
        e.stopPropagation();
      }}
      className="fixed bottom-4 right-4 z-[99999] shadow-lg"
      style={{
        zIndex: 99999,
        pointerEvents: 'auto',
      }}
      aria-label="Report a bug or give feedback"
    >
      <Bug className="h-4 w-4 mr-2" />
      Report an Issue
    </Button>
  );

  // Render the button in a portal to ensure it's outside modal context
  return createPortal(button, document.body);
} 