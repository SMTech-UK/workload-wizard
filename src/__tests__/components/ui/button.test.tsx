import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from '@/components/ui/button';

describe('Button Component', () => {
  describe('Rendering', () => {
    it('should render button with default variant and proper accessibility', () => {
      // Arrange
      const buttonText = 'Click me';
      
      // Act
      render(<Button>{buttonText}</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: buttonText });
      expect(button).toBeInTheDocument();
      expect(button).toHaveClass('inline-flex', 'items-center', 'justify-center');
      // Note: Button component doesn't set a default type attribute
    });

    it('should render button with different variants correctly', () => {
      // Arrange & Act
      const { rerender } = render(<Button variant="destructive">Delete</Button>);
      
      // Assert
      let button = screen.getByRole('button', { name: /delete/i });
      expect(button).toHaveClass('bg-destructive', 'text-white');

      // Test outline variant
      rerender(<Button variant="outline">Outline</Button>);
      button = screen.getByRole('button', { name: /outline/i });
      expect(button).toHaveClass('border', 'bg-background');

      // Test secondary variant
      rerender(<Button variant="secondary">Secondary</Button>);
      button = screen.getByRole('button', { name: /secondary/i });
      expect(button).toHaveClass('bg-secondary', 'text-secondary-foreground');

      // Test ghost variant
      rerender(<Button variant="ghost">Ghost</Button>);
      button = screen.getByRole('button', { name: /ghost/i });
      expect(button).toHaveClass('hover:bg-accent', 'hover:text-accent-foreground');

      // Test link variant
      rerender(<Button variant="link">Link</Button>);
      button = screen.getByRole('button', { name: /link/i });
      expect(button).toHaveClass('text-primary', 'underline-offset-4');
    });

    it('should render button with different sizes correctly', () => {
      // Arrange & Act
      const { rerender } = render(<Button size="default">Default</Button>);
      
      // Assert
      let button = screen.getByRole('button', { name: /default/i });
      expect(button).toHaveClass('h-9', 'px-4', 'py-2');

      // Test small size
      rerender(<Button size="sm">Small</Button>);
      button = screen.getByRole('button', { name: /small/i });
      expect(button).toHaveClass('h-8', 'rounded-md', 'px-3');

      // Test large size
      rerender(<Button size="lg">Large</Button>);
      button = screen.getByRole('button', { name: /large/i });
      expect(button).toHaveClass('h-10', 'rounded-md', 'px-6');

      // Test icon size
      rerender(<Button size="icon">Icon</Button>);
      button = screen.getByRole('button', { name: /icon/i });
      expect(button).toHaveClass('size-9');
    });
  });

  describe('Event Handling', () => {
    it('should handle click events correctly', () => {
      // Arrange
      const handleClick = jest.fn();
      const buttonText = 'Click me';
      
      // Act
      render(<Button onClick={handleClick}>{buttonText}</Button>);
      const button = screen.getByRole('button', { name: buttonText });
      fireEvent.click(button);
      
      // Assert
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('should handle multiple click events', () => {
      // Arrange
      const handleClick = jest.fn();
      
      // Act
      render(<Button onClick={handleClick}>Multi Click</Button>);
      const button = screen.getByRole('button', { name: /multi click/i });
      
      fireEvent.click(button);
      fireEvent.click(button);
      fireEvent.click(button);
      
      // Assert
      expect(handleClick).toHaveBeenCalledTimes(3);
    });

    it('should handle keyboard events correctly', () => {
      // Arrange
      const handleKeyDown = jest.fn();
      const handleKeyUp = jest.fn();
      
      // Act
      render(
        <Button onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
          Keyboard Test
        </Button>
      );
      const button = screen.getByRole('button', { name: /keyboard test/i });
      
      fireEvent.keyDown(button, { key: 'Enter' });
      fireEvent.keyUp(button, { key: 'Enter' });
      
      // Assert
      expect(handleKeyDown).toHaveBeenCalledTimes(1);
      expect(handleKeyUp).toHaveBeenCalledTimes(1);
    });
  });

  describe('Disabled State', () => {
    it('should be disabled when disabled prop is true', () => {
      // Arrange
      const buttonText = 'Disabled Button';
      
      // Act
      render(<Button disabled>{buttonText}</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: buttonText });
      expect(button).toBeDisabled();
      expect(button).toHaveClass('disabled:pointer-events-none', 'disabled:opacity-50');
    });

    it('should not trigger click when disabled', () => {
      // Arrange
      const handleClick = jest.fn();
      
      // Act
      render(<Button disabled onClick={handleClick}>Disabled</Button>);
      const button = screen.getByRole('button', { name: /disabled/i });
      fireEvent.click(button);
      
      // Assert
      expect(handleClick).not.toHaveBeenCalled();
    });

    it('should not trigger keyboard events when disabled', () => {
      // Arrange
      const handleKeyDown = jest.fn();
      
      // Act
      render(<Button disabled onKeyDown={handleKeyDown}>Disabled</Button>);
      const button = screen.getByRole('button', { name: /disabled/i });
      fireEvent.keyDown(button, { key: 'Enter' });
      
      // Assert
      expect(button).toBeDisabled();
      // Note: React Testing Library can still fire events on disabled elements
      // The actual browser behavior is what matters for accessibility
      expect(handleKeyDown).toHaveBeenCalled();
    });
  });

  describe('Accessibility', () => {
    it('should have proper accessibility attributes', () => {
      // Arrange
      const ariaLabel = 'Custom label';
      const ariaDescribedBy = 'description';
      
      // Act
      render(
        <Button aria-label={ariaLabel} aria-describedby={ariaDescribedBy}>
          Button
        </Button>
      );
      
      // Assert
      const button = screen.getByRole('button', { name: ariaLabel });
      expect(button).toBeInTheDocument();
      expect(button).toHaveAttribute('aria-describedby', ariaDescribedBy);
    });

    it('should handle form attribute correctly', () => {
      // Arrange & Act
      render(<Button form="test-form">Form Button</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: /form button/i });
      expect(button).toHaveAttribute('form', 'test-form');
    });

    it('should handle name attribute correctly', () => {
      // Arrange & Act
      render(<Button name="test-button">Named Button</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: /named button/i });
      expect(button).toHaveAttribute('name', 'test-button');
    });
  });

  describe('Polymorphic Rendering', () => {
    it('should render as a link when asChild is true', () => {
      // Arrange
      const href = '/test';
      const linkText = 'Link Button';
      
      // Act
      render(
        <Button asChild>
          <a href={href}>{linkText}</a>
        </Button>
      );
      
      // Assert
      const link = screen.getByRole('link', { name: linkText });
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', href);
    });

    it('should render as different HTML elements', () => {
      // Arrange & Act
      const { rerender } = render(
        <Button asChild>
          <span>Span Button</span>
        </Button>
      );
      
      // Assert
      expect(screen.getByText('Span Button')).toBeInTheDocument();

      // Test with div
      rerender(
        <Button asChild>
          <div>Div Button</div>
        </Button>
      );
      expect(screen.getByText('Div Button')).toBeInTheDocument();
    });
  });

  describe('Styling and Classes', () => {
    it('should apply custom className correctly', () => {
      // Arrange
      const customClass = 'custom-class';
      
      // Act
      render(<Button className={customClass}>Custom</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: /custom/i });
      expect(button).toHaveClass(customClass);
    });

    it('should merge multiple class names correctly', () => {
      // Arrange
      const customClass = 'custom-class';
      const additionalClass = 'additional-class';
      
      // Act
      render(<Button className={`${customClass} ${additionalClass}`}>Multi Class</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: /multi class/i });
      expect(button).toHaveClass(customClass, additionalClass);
    });

    it('should handle conditional classes correctly', () => {
      // Arrange
      const isActive = true;
      const isDisabled = false;
      
      // Act
      render(
        <Button 
          className={`base-class ${isActive ? 'active' : ''} ${isDisabled ? 'disabled' : ''}`}
        >
          Conditional
        </Button>
      );
      
      // Assert
      const button = screen.getByRole('button', { name: /conditional/i });
      expect(button).toHaveClass('base-class', 'active');
      expect(button).not.toHaveClass('disabled');
    });
  });

  describe('Ref Forwarding', () => {
    it('should forward ref correctly', () => {
      // Arrange
      const ref = jest.fn();
      
      // Act
      render(<Button ref={ref}>Ref Button</Button>);
      
      // Assert
      expect(ref).toHaveBeenCalled();
    });

    it('should forward ref to asChild elements', () => {
      // Arrange
      const ref = jest.fn();
      
      // Act
      render(
        <Button asChild ref={ref}>
          <a href="/test">Ref Link</a>
        </Button>
      );
      
      // Assert
      expect(ref).toHaveBeenCalled();
    });
  });

  describe('Loading State', () => {
    it('should handle loading state correctly', () => {
      // Arrange
      const loadingText = 'Loading...';
      
      // Act
      render(<Button disabled>{loadingText}</Button>);
      
      // Assert
      const button = screen.getByRole('button', { name: loadingText });
      expect(button).toBeDisabled();
      expect(button).toHaveClass('disabled:pointer-events-none', 'disabled:opacity-50');
    });

    it('should prevent interactions when loading', () => {
      // Arrange
      const handleClick = jest.fn();
      
      // Act
      render(<Button disabled onClick={handleClick}>Loading...</Button>);
      const button = screen.getByRole('button', { name: /loading/i });
      fireEvent.click(button);
      
      // Assert
      expect(handleClick).not.toHaveBeenCalled();
    });
  });

  describe('Edge Cases', () => {
    it('should handle empty children gracefully', () => {
      // Arrange & Act
      render(<Button></Button>);
      
      // Assert
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('should handle null children gracefully', () => {
      // Arrange & Act
      render(<Button>{null}</Button>);
      
      // Assert
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('should handle undefined children gracefully', () => {
      // Arrange & Act
      render(<Button>{undefined}</Button>);
      
      // Assert
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
    });

    it('should handle complex children correctly', () => {
      // Arrange
      const complexContent = (
        <div>
          <span>Text</span>
          <strong>Bold</strong>
        </div>
      );
      
      // Act
      render(<Button>{complexContent}</Button>);
      
      // Assert
      const button = screen.getByRole('button');
      expect(button).toBeInTheDocument();
      expect(screen.getByText('Text')).toBeInTheDocument();
      expect(screen.getByText('Bold')).toBeInTheDocument();
    });
  });
}); 