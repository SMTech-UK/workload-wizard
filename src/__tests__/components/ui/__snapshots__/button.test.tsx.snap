// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different sizes: button-default-size 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Default Size
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different sizes: button-icon-size 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 size-9"
    data-slot="button"
  >
    Icon Size
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different sizes: button-large-size 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-10 rounded-md px-6 has-[>svg]:px-4"
    data-slot="button"
  >
    Large Size
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different sizes: button-small-size 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5"
    data-slot="button"
  >
    Small Size
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-default-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Default Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-destructive-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-destructive text-white shadow-xs hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Destructive Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-ghost-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Ghost Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-link-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary underline-offset-4 hover:underline h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Link Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-outline-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Outline Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure across different variants: button-secondary-variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
  >
    Secondary Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure with custom className: button-custom-styling 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 bg-purple-500 hover:bg-purple-600 text-white font-bold"
    data-slot="button"
  >
    Custom Styled Button
  </button>
</div>
`;

exports[`Button Component Snapshot Testing should maintain consistent UI structure with disabled state: button-disabled-state 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3"
    data-slot="button"
    disabled=""
  >
    Disabled Button
  </button>
</div>
`;
